### PR TITLE
🚧 BUU: Pagination tweaks

### LINKED ISSUE #10878: [BUU] Pagination tweaks
## Context
This issue polishes the UI for https://github.com/openfoodfoundation/openfoodnetwork/issues/10877

## Tasks

- [x] increase padding top to be 20px
- [x] increase padding bottom to be 28px
- [x] increase distance between arrows and numbers to be 36px
- [x] increase distance between numbers to be 8px
- [x] font weight to be 600
- [x] change not selected font color to $near-black, also for arrows
- [x] update on hover as per designs attached
- [x] remove white background from dots as per designs (distance from previous and next should be 8px)
- [x] hide arrow buttons for when first or last page are loaded
- [x] hide pagination for single page catalogue
- [x] following [Pagy's docs](https://ddnexus.github.io/pagy/docs/how-to/#control-the-page-links), I'd say to change the size variable to [1,2,2,1], at the moment it becomes extremely long for big catalogues, see screenshot below with the default [1,4,4,1]

Screenshot 2023-08-07 at 16 31 40

Question: are we using the Fontawesome icons for the arrows? It's a bit hard to tell.

## Design files
https://www.figma.com/file/v1zbrWDZSRd3Nqoe0SJ2Sm/Engineering-Delivery---Back-Office?type=design&node-id=550%3A4374&mode=design&t=5U46fvJD0fG8K6FF-1

### PR DESCRIPTION
**Pull request by jibees**
#### What? Why?

- Closes #10878 

Capture d’écran 2023-08-15 à 11 52 32
Capture d’écran 2023-08-15 à 11 52 25
Capture d’écran 2023-08-15 à 11 52 20
Capture d’écran 2023-08-15 à 11 52 08
Capture d’écran 2023-08-15 à 11 52 02

#### What should we test?

- Visit ... page.
- 

#### Release notes

Changelog Category: User facing changes under feature toggle

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Review by mkllnk — APPROVED** [review]
Nice! :sparkle: 

You omitted testing notes but I think that we can skip testing here anyway.

File: app/webpacker/css/admin_v3/components/pagination.scss [hunk 1]
```diff
@@ -34,8 +35,8 @@
       margin-left: 28px; // 28+8 (the gap) = 36px : the space between the arrows and the numbers
     }
 
-    &.disabled {
-      cursor: default;
+    &:not(.disabled):not(.current) {
+      cursor: pointer;
```
Comment (dacook) [inline app/webpacker/css/admin_v3/components/pagination.scss:45]: I'm thinking we shouldn't need to specify the cursor at all.
If it has a `href`, it is clickable and the browser will decide to show the pointer. 
If it is disabled, then it should have no `href`. I see that currently, these `.disabled` elements have `href="#"`, but can this be removed?

File: app/webpacker/css/admin_v3/components/pagination.scss [hunk 2]
```diff
@@ -39,10 +39,16 @@
       cursor: pointer;
     }
 
-    &:hover:not(.disabled) {
+    &:hover:not(.disabled):not(.gap) {
```
Comment (dacook) [inline app/webpacker/css/admin_v3/components/pagination.scss:48]: Building upon my previous comment, how about we just scope it to clickable links? 
```suggestion
    &[href] {
```

File: app/views/admin/shared/v3/_pagy.html.haml [hunk 3]
```diff
@@ -2,9 +2,6 @@
   - if pagy.prev
     %a.page.prev{ href: "#", id: "pagy-prev", "data-reflex": reflex, "data-perPage": pagy.items, "data-page": pagy.prev || 1, "aria-label": "previous"}
       %i.icon-chevron-left
-  - else
-    %span.page.prev.disabled
```
Comment (dacook) [inline app/views/admin/shared/v3/_pagy.html.haml:6]: I guess the `.disabled` css rules can also be removed now.

**Review by dacook — CHANGES_REQUESTED** [review]
Great, it's looking good, though I noticed some unused CSS got left behind. 

I also have a suggestion for simplifying the markup and style rules, but it's optional ;)

File: app/webpacker/css/admin_v3/components/pagination.scss (hunk 1 above)
Comment (jibees) [reply app/webpacker/css/admin_v3/components/pagination.scss]: > If it has a href, it is clickable and the browser will decide to show the pointer.

Yes, but...
For the `current` element, it has a `text` cursor, so I need to set it as `default`. And then, I should set the not current|disabled to pointer. Reading this I could do also the exact opposite: setting to default only for current|disabled and let the pointer to be set as it's a link. Which is maybe easier to understand. 

Commit cfe79b8b79

File: app/webpacker/css/admin_v3/components/pagination.scss (hunk 2 above)
Comment (jibees) [reply app/webpacker/css/admin_v3/components/pagination.scss]: Yes! ad3f9001c3

File: app/views/admin/shared/v3/_pagy.html.haml (hunk 3 above)
Comment (jibees) [reply app/views/admin/shared/v3/_pagy.html.haml:6]: Oh, I guess so ... Thanks!

File: app/webpacker/css/admin_v3/components/pagination.scss [hunk 4]
```diff
@@ -14,23 +18,42 @@
...
     @include defaultBoxShadow;
     border-radius: 4px;
-    color: $color-9;
+    color: $color-8;
+    font-weight: 600;
+
+    &.current,
+    &.gap {
+      cursor: default;
+    }
```
Comment (dacook) [inline app/webpacker/css/admin_v3/components/pagination.scss:27]: As the `.cursor` and `.gap` elements are only `span`s, I don't think this rule is needed. Maybe something for the next PR :)

**Review by dacook — APPROVED** [review]
Great, let's merge!

**Pull request merged**